app/models/deploy.rb

```diff
@@ -179,8 +206,12 @@ def validate_stage_uses_deploy_groups_properly
     end
   end
 
-  def deploy_buddy
-    return unless stage.deploy_requires_approval?
+  def deploy_buddy(with_deleted: false)
+    if with_deleted
+      return unless stage_with_deleted.deploy_requires_approval?
```

Comment:
removed with_deleted: with_deleted option since to have valid information it should pull deleted stages.